pkgs/top-level/all-packages.nix

```diff
@@ -19114,6 +19114,8 @@ with pkgs;
 
   vault = callPackage ../tools/security/vault { };
 
+  vaultenv = callPackage ../tools/security/vaultenv { };
```

Comment:
Haskell packages need to be registered in `pkgs/development/haskell-modules/configuration-common.nix` instead. Look for `tinc` in that file for an example of how this is done. You'll find it's actually easier than the construct you've used now.